### PR TITLE
Jw/sc 4520/edit staff modal

### PR DESCRIPTION
**Pull request by wittejm**
https://user-images.githubusercontent.com/2104990/187553974-0e4fdff0-c15a-422d-8c65-66bfacd401ff.mov

### CODE REVIEW

File: src/components/System/AddEditStaffModal.tsx [hunk 1]
```diff
@@ -27,7 +31,10 @@ const AddStaffModal = ({ onClose, staff, systemId }: Props) => {
 
   const [createStaff] = useMutation(CreateStaffDocument, {
     onError: (error) => {
-      if (error.graphQLErrors[0].extensions.code === "BAD_USER_INPUT") {
+      if (
+        error.graphQLErrors[0] &&
+        error.graphQLErrors[0].extensions.code === "BAD_USER_INPUT"
+      ) {
         showToast("CreateStaff", error.message, "error");
       } else reportError(error, "CreateStaff");
```
Comment (cainlevy) [inline src/components/System/AddEditStaffModal.tsx:39]: oh hey i just merged an improvement to `reportError` that may make this unnecessary: https://github.com/Ameelio/connect-doc-client/pull/398

File: src/components/System/AddEditStaffModal.tsx [hunk 2]
```diff
@@ -43,37 +50,66 @@ const AddStaffModal = ({ onClose, staff, systemId }: Props) => {
...
   });
+  const [updateStaff] = useMutation(UpdateStaffDocument, {
+    onError: (error) => {
+      if (
+        error.graphQLErrors[0] &&
+        error.graphQLErrors[0].extensions.code === "BAD_USER_INPUT"
+      ) {
+        showToast("UpdateStaff", error.message, "error");
+      } else reportError(error, "UpdateStaff");
+    },
```
Comment (cainlevy) [inline src/components/System/AddEditStaffModal.tsx:61]: might simplify with https://github.com/Ameelio/connect-doc-client/pull/398

**Review by cainlevy — APPROVED** [review]

**Comment by cainlevy** [comment]
would you also run this past Liz?

**Comment by wittejm** [comment]
Will merge now and follow up with another PR for any changes.

**Comment by eliz214** [comment]
@wittejm slacked you some questions/feedback!

**Pull request merged**